pkg/ipcache/types.go

```diff
@@ -194,211 +245,202 @@ func (s *prefixInfo) sortedBySourceThenResourceID() []ipcachetypes.ResourceID {
-				overrideResourceID = resourceID
+				if out.identityOverride {
+					scopedLog.Warn(
+						"Detected conflicting identity override for prefix. "+
+							"This may cause connectivity issues for this address.",
+						logfields.Labels, out.labels,
+						logfields.Resource, overrideResourceID,
+						logfields.ConflictingLabels, info.labels,
+						logfields.ConflictingResource, resourceID,
+					)
+				} else {
+					out.identityOverride = true
+					out.labels = info.labels
+					overrideResourceID = resourceID
+				}
 			}
 		}
 
-		if info.tunnelPeer.IsValid() {
-			if tunnelPeer.IsValid() {
+		if info.tunnelPeer.IsValid() && info.tunnelPeer != out.tunnelPeer {
+			if out.tunnelPeer.IsValid() {
 				if option.Config.TunnelingEnabled() {
 					scopedLog.Warn(
 						"Detected conflicting tunnel peer for prefix. "+
 							"This may cause connectivity issues for this address.",
-						logfields.TunnelPeer, tunnelPeerResourceID,
-						logfields.Resource, resourceID,
+						logfields.TunnelPeer, out.tunnelPeer,
+						logfields.Resource, tunnelPeerResourceID,
 						logfields.ConflictingTunnelPeer, info.tunnelPeer,
 						logfields.ConflictingResource, resourceID,
 					)
 				}
 			} else {
-				tunnelPeer = info.tunnelPeer
+				out.tunnelPeer = info.tunnelPeer
 				tunnelPeerResourceID = resourceID
 			}
 		}
 
-		if info.encryptKey.IsValid() {
-			if encryptKey.IsValid() {
+		if info.encryptKey.IsValid() && info.encryptKey != out.encryptKey {
+			if out.encryptKey.IsValid() {
 				scopedLog.Warn(
 					"Detected conflicting encryption key index for prefix. "+
 						"This may cause connectivity issues for this address.",
-					logfields.Key, encryptKey,
+					logfields.Key, out.encryptKey,
 					logfields.Resource, encryptKeyResourceID,
 					logfields.ConflictingKey, info.encryptKey,
 					logfields.ConflictingResource, resourceID,
 				)
 			} else {
-				encryptKey = info.encryptKey
+				out.encryptKey = info.encryptKey
 				encryptKeyResourceID = resourceID
 			}
 		}
 
-		if info.requestedIdentity.IsValid() {
-			if requestedID.IsValid() {
+		if info.requestedIdentity.IsValid() && info.requestedIdentity != out.requestedIdentity {
+			if out.requestedIdentity.IsValid() {
 				scopedLog.Warn(
 					"Detected conflicting requested numeric identity for prefix. "+
 						"This may cause momentary connectivity issues for this address.",
-					logfields.Identity, requestedID,
+					logfields.Identity, out.requestedIdentity,
 					logfields.Resource, requestedIDResourceID,
-					logfields.ConflictingKey, info.requestedIdentity,
+					logfields.ConflictingIdentity, info.requestedIdentity,
 					logfields.ConflictingResource, resourceID,
 				)
 			} else {
-				requestedID = info.requestedIdentity
+				out.requestedIdentity = info.requestedIdentity
 				requestedIDResourceID = resourceID
 			}
 		}
 
-		if info.endpointFlags.IsValid() {
-			if endpointFlags.IsValid() {
+		// TODO: if there are ever more flags, merge them
```

Comment:
nit: leftover TODO?

Reply:
I think this comment probably belongs in `pkg/ipcache/types/types.go` instead, where the `EndpointFlags` structure is declared. Otherwise future writers adding a new endpoint flag may not fully appreciate that this function also needs to be updated.

I'd phrase this not as `TODO` (which implies that there is known work to do right now) but rather as an instruction to the developer changing `EndpointFlags` that they must also evaluate how a new flag should impact flattening of prefixInfo in `pkg/ipcache`.

Reply:
Unmarked as "resolved" as this comment was not yet resolved. Feel free to follow up on this in a separate PR if there's no other requested changes, but please do remove this. 

As-is the TODO will highly likely be overlooked by anyone attempting to make such a change. That's not to mention that TODOs almost always get completely ignored even if they're tagged with a specific owner. Generally speaking, if there's a TODO, we should either do it or resolve that the TODO is not important.

Reply:
(Reading further I see the suggestion above was taken, but the TODO is still here. I suggest just dropping the TODO.)
```suggestion
```